### ISSUE TITLE
Color Palette Critical issue in 6.1

### ISSUE THREAD

**Issue by mdmag** [WordPress/gutenberg#45569]
### Description

We were using ColorPalette component which we import from @wordpress/components
colors prop have been optional until 6.1 release
6.1 release crashes our app because it suppose that color prop is required.

### Step-by-step reproduction instructions

1- Import ColorPalette from "@wordpress/components"
2- Don't pass colors prop
3- You will notice a javascript error 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by skorasaurus** [comment]
Hi,

Thanks for reporting; could you provide a code example of with the error occurring as well as the error message as well ; I am not able to reproduce any error on a test case where I have the following in my edit.js 

````
import { ColorPalette } from "@wordpress/components";
import { useState } from "@wordpress/element";


const MyColorPalette = () => {
	const [color, setColor] = useState("#f00");

	return <ColorPalette value={color} onChange={(color) => setColor(color)} />;
};
````

**Comment by ndiego** [comment]
@ciampo I see you have done a decent amount of work on the ColorPalette component in the past few months. Any ideas on this one?

**Comment by Mamaduka** [comment]
Here's PR that aims to fix issue #45566.

I think we can use optional chaining as proposed in PR or set the default for `colors` params, as it's [documented](https://developer.wordpress.org/block-editor/reference-guides/components/color-palette/#colors-paletteobject-colorobject).

Cc @mirka

**Comment by mirka** [comment]
Ah, I think I understand what's going on now! The Gutenberg code in WP 6.1 is not up to date with our _documentation_, which is always in line with trunk.

In trunk, `colors` is documented as optional and it will not crash. In WP 6.1, `colors` is still required and will crash.

This change happened in dbbcc6eff90647ad3e41fe849932ce1b7fe6a46d. So if you check the commit before that (6ee2a495af64d730ac83d39987a8b8b8e09cdf66), the docs say that `colors` is required, and will crash.

**Comment by Mamaduka** [comment]
@mirka, the missing required `colors` param wasn't crashing in 6.0 because we were using `lodash.map`, which won't trigger an error if a variable is undefined.

https://github.com/WordPress/gutenberg/blob/release/6.0/packages/components/src/color-palette/index.js#L29

Should we consider this as a breaking change?

I understand that it was documented as required, but people often ignore docs.

**Comment by lynnbgriffith** [comment]
@tyxla This looks like a case where lodash was maybe being overly "forgiving" with types previously and covering up the issue. We should probably guard against this case to preserve the older behavior.

**Comment by tyxla** [comment]
Thanks for the ping! I'm going to take a look shortly.

**Comment by mirka** [comment]
> Should we consider this as a breaking change?

Hard to say in this specific case, but definitely the type of change we should be careful of when removing lodash dependencies!

As for this particular issue, it has been back to the previous behavior since #44632, with docs updated to mark the prop as optional. It should also be guarded against future regressions of the sort, thanks to it being in TypeScript now.

So the immediate action item here would be to include #44632 in the next release (not sure if there's a cherry picking situation going on for the patch releases).

**Comment by Mamaduka** [comment]
Thank you, @mirka!

I will add #44632 to the WP 6.1.1 board and cherry-pick it for the minor release.

**Comment by tyxla** [comment]
From what I understand, there isn't anything else to be done here, other than cherry-picking #44632. Unless y'all think we need additional undefined handling?

**Comment by Mamaduka** [comment]
That's my understanding as well, @tyxla.

**Comment by Mamaduka** [comment]
The fix was backported into WP 6.1.1. I'll close the issue.
